### PR TITLE
DYN-6173 add new close flag to splashscreen

### PR DESCRIPTION
**Pull request by mjkkirschner**
### Purpose

Adds a flag to the splashscreen so clients can tell why the splash screen was closed:
1. because dynamo was launched from the splash screen.
2. because the splash screen was closed.

- [x] add a test 

After this is merged, we'll publish a new nuget and consume it in this PR:
https://github.com/DynamoDS/DynamoRevit/pull/2955

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB 

### Release Notes

NA

### CODE REVIEW

**Review by QilongTang — APPROVED** [review]

**Comment by mjkkirschner** [comment]
@QilongTang PTAL I've fixed a bug, made some methods internal so I can use them during tests, and made splash screen work constructible without the DynamoView so it can be tested a bit in isolation.

File: src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs [hunk 1]
```diff
@@ -275,34 +275,52 @@ void DynamoView_MouseLeftButtonDown(object sender, MouseButtonEventArgs e)
...
-            
-            shortcutBar.IsNewButtonEnabled = enable;
-            shortcutBar.IsOpenButtonEnabled = enable;
-            shortcutBar.IsSaveButtonEnabled = enable;
-            shortcutBar.IsLoginMenuEnabled = enable;
-            shortcutBar.IsExportMenuEnabled  = enable;
-            shortcutBar.IsNotificationCenterEnabled = enable;
-
-            if(dynamoViewModel.ShowStartPage)
+            if (!(saveThisButton is null))
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs:278]: wrapped these in null checks, as this method get called during workspace shutdown, but if just the splash screen is visible these items are null and the splash screen is never closed.

File: src/DynamoCoreWpf/Views/SplashScreen/SplashScreen.xaml.cs [hunk 2]
```diff
@@ -209,11 +218,15 @@ private bool SignOut()
         /// <param name="isCheckboxChecked"></param>
         private void LaunchDynamo(bool isCheckboxChecked)
         {
-            viewModel.PreferenceSettings.EnableStaticSplashScreen = !isCheckboxChecked;
+            CloseWasExplicit = false;
+            if (viewModel != null)
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/Views/SplashScreen/SplashScreen.xaml.cs:222]: some null checks, as in my tests, I want to call these methods without the DynamoView actually being around.

File: src/DynamoCoreWpf/Views/SplashScreen/SplashScreen.xaml.cs [hunk 3]
```diff
@@ -468,10 +482,10 @@ private void CloseWindow()
             // If Dynamo is launched from an integrator host, user should be able to close the splash screen window.
             // Additionally, we will have to shutdown the ViewModel which will close all the services and dispose the events.
             // RequestUpdateLoadBarStatus event needs to be unsubscribed when the splash screen window is closed, to avoid populating the info on splash screen.
-            else if (this is SplashScreen)
```
Comment (mjkkirschner) [inline src/DynamoCoreWpf/Views/SplashScreen/SplashScreen.xaml.cs:471]: I don't really see how `this` would ever be null when this is called.

**Comment by mjkkirschner** [comment]
master-5 tests passed, I'm going to merge this.

**Pull request merged**